packages/blob/src/helpers.ts

```diff
@@ -32,12 +35,65 @@ export class BlobAccessError extends Error {
   }
 }
 
+export class BlobNotFoundError extends Error {
```

Comment:
Can we rename this to BlobStoreNotFoundError instead? and make sure `not_found` is `store_not_found` here and in the API.

Reply:
I see S3 throws when an object doesn't exist.
I see R2 doesn't throw when an object doesn't exist.

Since our methods can throw in general (for multiple reasons), I would be fine if we were to throw on object not found 👍 What do you think?

Reply:
I agree, throwing a specific error seems more meaningful for the user